### PR TITLE
Onboarding integration with auth

### PR DESCRIPTION
**Pull request by Polarpi**
## Summary

This PR integrates the Onboarding Wizard with the users in MongoDB creating with Google OAuth. When the user submits the onboarding information, the user in the database is updated.

Closes #57

### CODE REVIEW

File: api/src/api/users.ts [hunk 1]
```diff
@@ -80,11 +80,13 @@ router.put(
       return;
     }
 
-    const updatedUser = await User.findByIdAndUpdate(
-      req.params.userId,
-      req.body,
-      { new: true, runValidators: true },
-    );
+    const user = req.body;
+    Object.keys(user).forEach((key) => user[key] === null && delete user[key]);
```
Comment (alicesf2) [inline api/src/api/users.ts:84]: Whoa :0 what is this for, exactly?

File: client/src/pages/wizard/Onboard3.tsx [hunk 2]
```diff
@@ -34,18 +35,38 @@ const Onboard3: FC<IProps> = ({ races, setRaces }): ReactElement => {
 
   // All of the race buttons to show in the first row
   const raceButtonsRow1 = [
-    { value: 'American Indian or Alaskan Native', color: '#EF8B8B' },
-    { value: 'Black or African American', color: '#A5C4F2' },
-    { value: 'Middle Eastern or North African', color: '#BAB9E9' },
-    { value: 'Native Hawaiian or Pacific Islander', color: '#A9D3E5' },
+    {
+      display: 'American Indian or Alaskan Native',
```
Comment (alicesf2) [inline client/src/pages/wizard/Onboard3.tsx:39]: What's the purpose of adding separate display and value fields here? Is it so you don't have to convert the value to uppercase before storing it in the db?

File: client/src/pages/wizard/WizardWrapper.tsx [hunk 3]
```diff
@@ -146,29 +173,41 @@ const WizardWrapper = (): ReactElement => {
...
-      portfolio: portfolio,
-      linkedIn: linkedIn,
-      twitter: twitter,
-      claimedPitches: [],
-      submittedPitches: [],
-      currentTeams: currentTeams,
-      role: role,
-      races: races,
-      interests: interests,
+      firstName: firstName !== '' ? firstName : null,
```
Comment (alicesf2) [inline client/src/pages/wizard/WizardWrapper.tsx:176]: Why do we need to convert empty strings to nulls here?

File: client/src/pages/wizard/WizardWrapper.tsx [hunk 4]
```diff
@@ -146,29 +173,41 @@ const WizardWrapper = (): ReactElement => {
...
+        reasonForInvolvement !== '' ? reasonForInvolvement : null,
+      currentTeams: currentTeams !== [] ? currentTeams : null,
+      role: role !== '' ? role : null,
+      races: races !== [] ? races : null,
+      interests: interests !== [] ? interests : null,
     };
 
-    console.log(formData);
+    const userUrl = `${BASE_URL}/users/${user._id}`;
+    axios
```
Comment (alicesf2) [inline client/src/pages/wizard/WizardWrapper.tsx:194]: Can we extract this to the `apiWrapper` file?

File: client/src/pages/wizard/WizardWrapper.tsx [hunk 5]
```diff
@@ -300,7 +339,7 @@ const WizardWrapper = (): ReactElement => {
               setModalOpen={setOpenModal}
             />
           )}
-          {page === WizardPage.COMPLETION.toString() && <Compleition />}
+          {page === WizardPage.COMPLETION.toString() && <Completion />}
```
Comment (alicesf2) [inline client/src/pages/wizard/WizardWrapper.tsx:342]: lmao ty for catching this

**Review by alicesf2 — COMMENTED** [review]
Just leaving general comments for now because I have a few questions first 😛 but awesome work, Eugenia! 💯 🐈

File: client/src/pages/wizard/WizardWrapper.tsx [hunk 6]
```diff
@@ -60,13 +61,30 @@ const WizardWrapper = (): ReactElement => {
...
   const [scheduleConfirmed, setScheduleConfirmed] = useState<boolean>(false);
 
+  useEffect(() => {
+    const userUrl = `${BASE_URL}/auth/currentuser`;
+    axios
+      .get(userUrl, {
+        headers: {
+          'Content-Type': 'application/JSON',
+        },
+      })
```
Comment (dyang5200) [inline client/src/pages/wizard/WizardWrapper.tsx:75]: Maybe should put the api call into a function in apiWrapper.tsx?

**Review by dyang5200 — APPROVED** [review]

File: api/src/api/users.ts (hunk 1 above)
Comment (Polarpi) [reply api/src/api/users.ts:85]: It just removes the fields we don't want to update

File: client/src/pages/wizard/Onboard3.tsx (hunk 2 above)
Comment (Polarpi) [reply client/src/pages/wizard/Onboard3.tsx]: Mm kind of. I think it's just better to use the enum itself for the value instead of converting to uppercase

**Review by amit-sawhney — APPROVED** [review]

**Comment by amit-sawhney** [comment]
We just need to clean up the merge conflicts but I think this looks good to me :)

**Pull request closed without merging**